src/components/structures/MatrixChat.js

```diff
@@ -275,11 +279,31 @@ module.exports = React.createClass({
                     screen: 'post_registration'
                 });
                 break;
+            case 'start_login_from_guest':
```

Comment:
Does this need to be a separate thing? ie. could start_login just do `if (MatrixClientPeg.get().isGuest()) {...` and if something explicitly wants to log out first, it can just post the `logout` action first?